### PR TITLE
Add e2e tests without DB dependence

### PR DESCRIPTION
**Pull request by not4Pedro**
Now tests can be run:
- pytest 
- pytest --run-e2e
- pytest --run-e2e-ci

##### This PR as well introduces:
- A whole stack of core (with in memory DB) and GUI (from build) is started for E2E tests.
- Login page no longer requires reload to login, loading of store is fixed.
- Added separate workflow for E2E CI tests. They at the moment most likely fail.
- Locally, the E2E tests run fine.

## Summary by CodeRabbit

- **New Features**
	- Introduced selective end-to-end testing capabilities, allowing tests to be run only with a specific command-line option.
	- Enhanced the initialization process of the web application to set up various components and services asynchronously.

- **Refactor**
	- Simplified the context creation in end-to-end testing by removing unnecessary parameters.
	- Streamlined route configuration by removing specific permission checks from the web application's routing logic.

- **Tests**
	- Added a new test script for simulating user interactions in a web application through end-to-end testing.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -38,6 +38,45 @@ jobs:
...
+
+      - name: test
+        working-directory: src/core
+        run: pytest --run-e2e-ci
+        env:
+          API_KEY: supersecret
+          SECRET_KEY: supersecret
+          FLASK_DEBUG: 1
+          DEBUG: true
+          SQLALCHEMY_DATABASE_URI: "postgresql+psycopg2://taranis:supersecret@localhost/taranis"
```
Comment (b3n4kh) [inline .github/workflows/tests.yaml:75]: why not the inmemory sqlite ?

**Comment by b3n4kh** [comment]
There is a fixture to use the pytest server for e2e tests: https://pytest-flask.readthedocs.io/en/stable/features.html#live-server-application-live-server

File: src/gui/src/main.js [hunk 2]
```diff
@@ -12,39 +12,46 @@ import VueDatePicker from '@vuepic/vue-datepicker'
 import '@vuepic/vue-datepicker/dist/main.css'
 
 import * as Sentry from '@sentry/vue'
-
+import { useMainStore } from '@/stores/MainStore'
```
Comment (b3n4kh) [inline src/gui/src/main.js:15]: Why are those changes necessary?

File: src/gui/src/stores/AssessStore.js [hunk 3]
```diff
@@ -189,6 +189,7 @@ export const useAssessStore = defineStore(
           return
         }
         groupAction(storySelection.value)
+        console.log(storySelection.value)
```
Comment (b3n4kh) [inline src/gui/src/stores/AssessStore.js:192]: ?

File: src/core/tests/playwright/conftest.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+@pytest.fixture(scope="session")
+def temporary_taranis_instance():
+    subprocess.run(["chmod", "+x", "./tests/playwright/temporary_taranis_instance.sh", "&>/dev/null &"])
+    result = subprocess.run(["./tests/playwright/temporary_taranis_instance.sh", "up", "&>/dev/null &"])
+    if result.returncode != 0:
+        pytest.fail("Script failed")
+    yield result.stdout
+    subprocess.run(["./tests/playwright/temporary_taranis_instance.sh", "down"])
+
```
Comment (b3n4kh) [inline src/core/tests/playwright/conftest.py:18]: Could be replaced by a simple call to `npm run build` and directly served by via python, so only a single process needs to be running.

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -38,6 +38,31 @@ jobs:
...
+          key: ${{ hashFiles('src/core/pyproject.toml') }}
+
+      - name: Install dependencies
+        working-directory: src/core
+        run: pip install -e .[dev]
+
+      - name: test
+        working-directory: src/core
+        run: pytest --run-e2e-ci
+
```
Comment (b3n4kh) [inline .github/workflows/tests.yaml:65]: Should be its own pipeline, only running if there for `tags`

File: src/core/tests/playwright/conftest.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import subprocess
+import pytest
+import requests
+
+headers = {
+    "Authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmcmVzaCI6ZmFsc2UsImlhdCI6MTcxNDY3NTMzNywianRpIjoiZTA3NjRiZmQtMDUzOC00NDg4LTljZmMtY2NkZmFjZGM2MGU2IiwidHlwZSI6ImFjY2VzcyIsInN1YiI6ImFkbWluIiwibmJmIjoxNzE0Njc1MzM3LCJjc3JmIjoiZWRmOTc2NTktNzAxMy00NzViLTg3ZDUtNTM0NTczM2VmY2NlIiwidXNlcl9jbGFpbXMiOnsiaWQiOjEsIm5hbWUiOiJBcnRodXIgRGVudCIsInJvbGVzIjpbMV19fQ.siZID18vdHC1tyQnW8B9KuwqU-B2QJZHPBpknAyB-qg"
+}
+
```
Comment (b3n4kh) [inline src/core/tests/playwright/conftest.py:8]: remove and replace with fixture

**Review by b3n4kh — CHANGES_REQUESTED** [review]

File: src/gui/src/main.js (hunk 2 above)
Comment (not4Pedro) [reply src/gui/src/main.js:15]: there was a problem feeding the application, and you fixed it for me. The problem was with the login page that needed a reload (tested in "private windows")

Comment (not4Pedro) [reply src/gui/src/main.js:15]: you might want to check this commit https://github.com/taranis-ai/taranis-ai/pull/203/commits/f93466cda01f45534c2f3840ac63b379ef24b6c8

File: src/core/tests/playwright/conftest.py (hunk 4 above)
Comment (not4Pedro) [reply src/core/tests/playwright/conftest.py]: possibly resolved

**Comment by b3n4kh** [comment]
Obsoleted by: https://github.com/taranis-ai/taranis-ai/commit/3511ae215db53cff5e5cf2dc2b2c225ffb735ef0

**Pull request closed without merging**